apis/core.oam.dev/v1beta1/application_types.go

```diff
@@ -52,18 +52,20 @@ type ApplicationComponent struct {
 
 // AppPolicy defines a global policy for all components in the app.
 type AppPolicy struct {
+	// Name is the unique name of the policy.
+	Name string `json:"name"`
+
 	Type string `json:"type"`
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Properties runtime.RawExtension `json:"properties,omitempty"`
 }
 
 // WorkflowStep defines how to execute a workflow step.
 type WorkflowStep struct {
-	Type string `json:"type"`
+	// Name is the unique name of the workflow step.
+	Name string `json:"name"`
```

Comment:
do we need to provide some automatic way to generate the name? e.g the index number

Reply:
One more reason: the name is used to pass down cue context.